examples/deepcopy-gen/generators/deepcopy.go

```diff
@@ -546,66 +549,85 @@ func (g *genDeepCopy) doSlice(t *types.Type, sw *generator.SnippetWriter) {
 }
 
 func (g *genDeepCopy) doStruct(t *types.Type, sw *generator.SnippetWriter) {
-	if len(t.Members) == 0 {
-		// at least do something with in/out to avoid "declared and not used" errors
-		sw.Do("_ = in\n_ = out\n", nil)
-		return
-	}
-
 	if hasDeepCopyMethod(t) {
 		sw.Do("*out = in.DeepCopy()\n", nil)
 		return
 	}
 
+	// Simple copy covers a lot of cases.
+	sw.Do("*out = *in\n", nil)
+
+	// Now fix-up fields as needed.
 	for _, m := range t.Members {
 		t := m.Type
+		hasMethod := false
+		if hasDeepCopyMethod(t) {
+			hasMethod = true
+		}
 		if t.Kind == types.Alias {
 			copied := *t.Underlying
 			copied.Name = t.Name
 			t = &copied
 		}
 		args := generator.Args{
 			"type": t,
+			"kind": t.Kind,
 			"name": m.Name,
 		}
 		switch t.Kind {
-		case types.Builtin:
-			sw.Do("out.$.name$ = in.$.name$\n", args)
+		case types.Builtin, types.Array:
```

Comment:
I might be missing something about Go, but is adding Array here really a good idea? What if underlying type of an arrays is something complex that requires proper deep-copy?